core/src/main/java/com/linecorp/armeria/client/endpoint/OrElseEndpointGroup.java

```diff
@@ -18,25 +18,27 @@
 
 import static java.util.Objects.requireNonNull;
 
-import java.util.List;
+import java.util.Collection;
 
 import com.linecorp.armeria.client.Endpoint;
 
-final class OrElseEndpointGroup implements EndpointGroup {
+final class OrElseEndpointGroup extends EndpointGroup {
     private final EndpointGroup first;
     private final EndpointGroup second;
 
     OrElseEndpointGroup(EndpointGroup first, EndpointGroup second) {
         this.first = requireNonNull(first, "first");
         this.second = requireNonNull(second, "second");
+        updateEndpoints(null);
```

Comment:
It seems weird for the endpoints to be duplicated in the OrElse and the delegates.

What do you think about adding a generic ```Listenable``` interface, similar to this one?
http://netflix.github.io/curator/doc/com/netflix/curator/framework/listen/Listenable.html

And this could be something like 

```java
class OrElseEndpointGroup implements Listenable<Collection<Endpoint>>, EndpointGroup {



first.addListener(() -> notifyListeners(endpoints()));
second.addListener(() -> notifyListeners(endpoints()));
```

Could also have ```AbstractListenable``` that takes care of the listener state management, but could potentially be hard to insert into inheritance chains.